imgui-SFML.cpp

```diff
@@ -114,15 +115,7 @@ int closeKeyboardIME()
 #endif
 #endif
 
-// Supress warnings caused by converting from uint to void* in pCmd->TextureID
-#ifdef _MSC_VER
-#pragma warning(push)
-#pragma warning(disable:4312)
-#elif __clang__
-#pragma clang diagnostic ignored "-Wint-to-void-pointer-cast" // warning : cast to 'void *' from smaller integer type 'int'
-#elif defined(__GNUC__)
-#pragma GCC diagnostic ignored "-Wint-to-pointer-cast"      // warning: cast to pointer from integer of different size
-#endif
+static_assert(sizeof(GLuint) <= sizeof(ImTextureID), "ImTextureID is not large enough to fit GLuint.");
```

Comment:
static_assert is a C++11 feature... This repo is using C++03. I think it'll be okay to put this assert under a macro which checks if we're in C++>=11 and to a runtime assert in ImGui::SFML::Init